src/transaction.js

```diff
@@ -94,6 +94,18 @@ class Transaction {
                 if (!Number.isSafeInteger(foreignAssetIndex) || foreignAssetIndex < 0) throw Error("each foreign asset index must be a positive number and smaller than 2^53-1");
             });
         }
+        if (assetMetadataHash !== undefined && assetMetadataHash.length !== 0) {
+            if (typeof(assetMetadataHash) === 'string') {
+                const encoded = Buffer.from(assetMetadataHash);
+                if (encoded.byteLength !== 32) {
+                    throw Error("assetMetadataHash must be a 32 byte Uint8Array or string.");
+                }
+                assetMetadataHash = new Uint8Array(encoded);
+            } else if (assetMetadataHash.constructor !== Uint8Array || assetMetadataHash.byteLength !== 32)
+                throw Error("assetMetadataHash must be a 32 byte Uint8Array or string.");
+        } else {
+            assetMetadataHash = undefined;
+        }
```

Comment:
The comment of the original function specifies that it needs to be a string. We should add a comment that specifies that it must be 32 bytes long. 
Also, since we specify it's a string, consider accepting only that (I guess with strict b64 form?).